core/sql/generator/GenRelScan.cpp

```diff
@@ -214,22 +215,32 @@ int HbaseAccess::createAsciiColAndCastExpr(Generator * generator,
       ((CharType*)newGivenType)->setUpshifted(FALSE);
     }
 
+  if (newGivenType->getTypeQualifier() == NA_CHARACTER_TYPE &&
+      CmpCommon::getDefaultString(HIVE_FILE_CHARSET) == "GBK")
```

Comment:
This may need to upshift the default value first, to catch the case where the user sets the CQD to lower case 'gbk'?